app/feature/feature-forever/src/main/kotlin/com/hedvig/android/feature/forever/ui/ReferralList.kt

```diff
@@ -0,0 +1,183 @@
+package com.hedvig.android.feature.forever.ui
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.LocalTextStyle
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material3.Divider
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.apollo.format
+import com.hedvig.android.core.designsystem.material3.typeElement
+import com.hedvig.android.core.designsystem.material3.warningElement
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.ui.getLocale
+import com.hedvig.android.feature.forever.ForeverUiState
+import hedvig.resources.R
+import javax.money.CurrencyContext
+import javax.money.CurrencyUnit
+import org.javamoney.moneta.Money
+
+@Composable
+internal fun ReferralList(uiState: ForeverUiState) {
+  if (uiState.referrals.isEmpty()) {
```

Comment:
Yeah this works. It also works to have on the call site instead something like
```
if (uiState.referrals.isNotEmpty()) {
  ReferralList(uiState)
}
```
So that it's not called in the first place at all.
I don't know if there's any *good* reason to chose one over the other in this particular situation, it may just be a stylistic choice, in which case do whatever feels right to you.

The one scenario where doing the if check on the call site would be for sure better is if we need to know there that if it shows or not, something else should change. For example
```
@Composable
fun Foo() {
  SomeText()
  if (someList.isNotEmpty()) {
    ReferralsList()
  } else {
    Spacer()
    Divider()
  }
}
```
Or something like that.

In that case it's important for it to be on the call site, since the alternative would be
```
@Composable
fun Foo() {
  SomeText()
  ReferralsList()
  if (someList.isEmpty()) {
    Spacer()
    Divider()
  }
}
```
And it would not be immediately obvious that if the list is not empty ReferralsList() shows, but if it's empty only then the spacer + divider shows.

I hope I am getting my point across here :D

Reply:
Thats a good point. I also think its good to know on the call site if the function will actually run or not in this case.